var/spack/repos/builtin/packages/lhapdfsets/package.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import os
+
+from spack import *
+
+
+class Lhapdfsets(BundlePackage):
+    """A set of disretised data files of parton density functions ,
+       to be used with the LHAPDF library"""
+
+    homepage = "https://lhapdf.hepforge.org/pdfsets.html"
+
+    tags = ['hep']
+
+    maintainers = ['vvolkl']
+
+    version('6.3.0')
+
+    depends_on('lhapdf', type='build')
+
+    phases = ['install']
+
+    # use a dummy executables for spack external support
+    executables = [r'^lhapdf$']
+
+    # parse set names from index file
+    all_sets = [_line.split()[1] for _line in open(join_path(os.path.dirname(__file__),
+                                                   'pdfsets.index')).readlines()]
+    default_sets = ["MMHT2014lo68cl", "MMHT2014nlo68cl", "CT14lo", "CT14nlo"]
+
+    variant('sets', description="Individiual lhapdf sets to install", values=disjoint_sets(('all',), ('default',), all_sets).with_default('default'))
```

Comment:
Do we really need a variant with more than 1k possible values? How does it look like on `spack info`?

Reply:
Sure, basically this package is very similar to [geant4-data](https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/geant4-data/package.py) - it's a collection of data files needed for several HEP generators on runtime, so it really is a dependency of those.  The only difference is that there are so many datasets in this one - they could probably be bundled more by the authors but they aren't, so we are stuck with them.

There are quite a few packages that need those, currently:
```
apfel  genie     pythia8    syscalc  thepeg  whizard
dire   herwig3  madgraph5amc   superchic  tauola   vbfnlo
```
and they all need different subsets. I agree that the list of variants is not pretty or readable, but it does give a lot of flexibility when adding a new generator.

However, if there is some overhead associated with this many variants, I'm also open to restricting them more to a few  sets coinciding with the depencency lists of the above generators (which then might mean that this recipe needs to see more maintenance as additional ones are added)

Reply:
@alalazo So, do you prefer to reduce the number of variants?

Reply:
Hi @vvolkl , @alalazo ,
I am not sure that this is a good idea to have, so called, "lhapdfsets" package in spack...
In CERN SFT LCGCMake we added this "package" with few sets in order to be able to compile "herwig++" which performs several checks at configure or build time(I do not remember exactly when). It was not used later at any stage by users or by LHC experiments. Probably what you have currently in spack for herwig++ recipe was based on LCGCmake recipe copied by Ivan (@iarspider).
And this was not quite wise idea as I see it now. Probably the best was to patch herwig++ to avoid this check or to communicate with asking them to remove such check.
Yes, at runtime you will need some PDF sets but this depends on your physics configuration. 

Therefore I do not see any reason why we need to change(change hash) LHAPDF and all downstream packages when you change what you distribute inside "lhapdfsets". 

The case differs from the case of Geant4 and its data files where we have clear correspondence (1:1) of data files and Geant4 versions.

And then, how do you want to assign versions to it as pdfsets.index is changing with every update of pdf sets?

The LHC experiments have their own copies and do their own versioning for reproducibility.

Reply:
Hi @dkonst13,

thanks for looking into this, and I completely agree with the issues you mention. If the need to access the lhapdfsets in herwig can be removed, that would be great.

However, I still think it makes sense to have this package in spack (we can even remove the individual sets from the variant list as I proposed above).

First, since  they are a runtime dependency, they may still be needed for testing.

Second, the usage I foresee for lcg and experiments is to declare this as an "external" package and keep using the existing installations. This will then just allow them to minimize the maintenance, as the `LHAPDF_DATA_PATH` variable can be set up with spack.

And if someone needs to install generators who need these sets from scratch, it will also be easier with these packages.

So I'm very happy if things can be simplified in the lcg releases, but to install the current model having this package at the moment is the best option, imo.

Reply:
Hi @vvolkl, 

Okay, we can have it as a separate package. But then nothing should depend on it, and it should not depend on anything else. And then ,of course, there are questions : how are you going to assign versions to it? And same question about variants. Usually people use just small subset of it.